File: scala/caesar.scala
// run with scala caesar.scala
object Caesar extends App {

    println(encrypt("ATTACK AT ONCE", 4))
    println(decrypt("EXXEGO EX SRGI", 4))

    solve("abcdeFGHIJKLmnopqrstuvwxyz ,?;{[()]}", 26)

    def encrypt(input: String, shift: Int): String = {
        var result = ""
        for(i <- input.toUpperCase) {
            if(!i.isUpper) {
                result += i
            } else {
                val n = (i - 65 + shift) % 26 + 65
                val output = n.toChar
                result += output
            }
        }
        result
    }

    def decrypt(input: String, shift: Int): String = {
        encrypt(input, 26-shift)
    }

    def solve(input: String, maxShift: Int): Unit = {
        for(i <- 0 to maxShift) {
            val output = decrypt(input, i)
            println(i + "\t" + output)
        }
    }

}

File: pascal/caesar.pas
{compile and run with fpc -v0 caesar.pas && ./caesar}
program Hello;
uses sysutils;
function encrypt(str:string; shift:integer): string;
var i: integer;
begin
    str:=upcase(str);
    for i:=1 to length(str) do
        begin
            {test for if it is a letter, at this point all is uppercase}
            if (str[i] in ['A'..'Z']) then
                begin
                    str[i]:= chr((ord(str[i]) -65 + shift) mod 26 + 65);
                end;
        end;
    encrypt:=str;
end;

function decrypt(str:string; shift:integer): string;
var i: integer;
begin
    decrypt:=encrypt(str, 26-shift);
end;

procedure solve(str:string; maxShiftValue:integer);
var i: integer;
begin
    str:=upcase(str);
    for i:=1 to 26 do
        begin
            writeln('Shift: ', i, ' Result: ', decrypt(str, i));
        end;
end;

begin
    writeln(encrypt('ATTACK AT ONCE',4));
    writeln(decrypt('EXXEGO EX SRGI',4));
    solve('abcdeFGHIJKLmnopqrstuvwxyz ,?;{[()]}', 26)
end.



